### PR TITLE
Remove "help" and "version" commands

### PR DESCRIPTION
**Pull request by lucperkins**
There are two things that I found a bit confusing about the VK CLI tool:

* If you try to get the version by running `virtual-kubelet version` you will get an error because you've failed to specify a provider using the `--provider` flag
* There's a `virtual-kubelet help` command that doesn't currently work because there are no commands, only flags

This PR does three things:

* It removes the `version` command and replaces it with a `--version` flag (which now does not throw an error when invoked)
* It removes the `help` command altogether (the `--help` flag is still available)
* It removes the check for the `--provider` flag into the `Run` function, which enables you to run `virtual-kubelet --help` without getting an error.

## Testing

To test locally (assuming that `$GOPATH/bin` is in the `PATH`):

```bash
go install
virtual-kubelet --help
# Displays help text
virtual-kubelet --version
# Virtual Kubelet: N/A
virtual-kubelet
# Throws an error because no provider is specified
```

### CODE REVIEW

**Comment by lucperkins** [comment]
I'm not sure how to interpret the CI failure. If someone from the team could provide me some insight, that'd be great.

**Comment by pires** [comment]
@robbiezhang I think we should remove ACI calls from PR checks as it's a source for flakiness.

**Comment by cpuguy83** [comment]
I think the main issue here may be incorrect the use of `PersistentFlags` and an improper command structure.

File: cmd/root.go [hunk 1]
```diff
@@ -165,6 +177,13 @@ func (mv mapVar) Type() string {
 func init() {
 	cobra.OnInitialize(initConfig)
 
+	RootCmd.SetHelpCommand(&cobra.Command{
+		Use:    "hidden-help-command",
```
Comment (cpuguy83) [inline cmd/root.go]: This still doesn't work.

**Comment by cpuguy83** [comment]
This whole package needs to be re-written... it's terrible.

Looks like this is really where the issue with the provider being set is:
```go
func init() {
	cobra.OnInitialize(initConfig)
```

We're doing validation on every command execution.

**Comment by cpuguy83** [comment]
Moving the call to `initConfig` into `Run` fixes the problem... this is from master with that:

```
bin/virtual-kubelet version
Version: v0.7.4-43-g20911aa3-dev, Built: 2019-01-15-19:07 UT
```

File: cmd/root.go (hunk 1 above)
Comment (lucperkins) [reply cmd/root.go]: The purpose of this, though, is the *remove* the `help` command in favor of the `--help` flag. As far as I can tell this does accomplish that.

**Comment by lucperkins** [comment]
@cpuguy83 I don't know about *terrible* but yeah, there are some warts. I'll take another look and see if there's a better general approach.

**Comment by cpuguy83** [comment]
I fixed these issues as part of #527 
Thanks! 🙇

**Pull request closed without merging**